### ISSUE TITLE
Cannot correctly read remote hostname / ip pairs

### ISSUE THREAD

**Issue by aq1018** [cassandra-rb/cassandra#116]
I have a 4 node cluster running `Cassandra 8.4`. 
The gem I'm using is `cassandra-0.12.1`. 
Ruby version is `ruby-1.9.2-p180`

The servers are all in local network with the following domain and ip:

```
ip           hostname
10.3.170.61  cass1
10.3.170.62  cass2
10.3.170.63  cass3
10.3.170.64  cass4
```

Also, `/etc/hosts` of the nodes is setup to resolve their host name to the IP described above.

I have the default 'Twitter' schema but tweaked to use `SimpleStrategy` as the placement strategy with replication factor set to 3.

Using another machine not on the node:

```
require 'cassandra'

# How to get exception on connect
c = Cassandra.new 'Twitter', 'cass1'
c.keyspaces
# Exception
CassandraThrift::Cassandra::Client::TransportException: CassandraThrift::Cassandra::Client::TransportException


# After exception, I saw the servers are being auto discovered.
# But notice the server's addresses are hostname/ipaddress:port, instead of just ipaddress:port
# Also, notice that I connected to cass1, and that returned 10.3.170.61:9160 with no hostname
#
c
 => #<Cassandra:26202580, @keyspace="Twitter", @schema={}, @servers=["cass2/10.3.170.62:9160", "cass3/10.3.170.63:9160", "cass4/10.3.170.64:9160", "10.3.170.61:9160"]> 

# Drilling into the code trying to confirm the format of the servers are causing the problems
t = ThriftClient.new(CassandraThrift::Cassandra::Client, "10.3.170.61:9160")
t.describe_ring 'Twitter' # This works because the address is correct
=> [<CassandraThrift::TokenRange start_token:"0", end_token:"42535295865117307932921825928971026432", endpoints:["cass2/10.3.170.62", "cass3/10.3.170.63", "cass4/10.3.170.64"]>, <CassandraThrift::TokenRange start_token:"42535295865117307932921825928971026432", end_token:"85070591730234615865843651857942052864", endpoints:["cass3/10.3.170.63", "cass4/10.3.170.64", "10.3.170.61"]>, <CassandraThrift::TokenRange start_token:"85070591730234615865843651857942052864", end_token:"127605887595351923798765477786913079296", endpoints:["cass4/10.3.170.64", "10.3.170.61", "cass2/10.3.170.62"]>, <CassandraThrift::TokenRange start_token:"127605887595351923798765477786913079296", end_token:"0", endpoints:["10.3.170.61", "cass2/10.3.170.62", "cass3/10.3.170.63"]>]

t = ThriftClient.new(CassandraThrift::Cassandra::Client, "cass1/10.3.170.61:9160")
t.describe_ring 'Twitter'
# Exception
CassandraThrift::Cassandra::Client::TransportException: CassandraThrift::Cassandra::Client::TransportException
```

Is my configuration of cassandra wrong? Is there a way to configure cassandra so that instead of returning hostname / ipaddress pairs, I can configure it so only IP or host name is returned? Or this is a bug in this gem?

**Comment by fuentesjr** [comment]
I'm having this issue as well. Sounds like its a bug introduced in 0.8.4. 

tl;dr: disable node auto discovery for now

Have a look at the following tickets for details:

https://issues.apache.org/jira/browse/CASSANDRA-3103
https://issues.apache.org/jira/browse/CASSANDRA-3044

**Comment by fuentesjr** [comment]
Since this is fixed in 0.8.5 and that should be out soon (http://mail-archives.apache.org/mod_mbox/cassandra-user/201109.mbox/%3CCAMXPLPgpt0DZgNpPgjXfQb5b-n5bxwy4fTCeS6awdFKi+stFbA@mail.gmail.com%3E), might just want to wait for the 0.8.5 if possible.
